// @uidotdev/react-query-api/dist/react-query-api.cjs.development.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var msw = require('msw');

function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) {
  try {
    var info = gen[key](arg);
    var value = info.value;
  } catch (error) {
    reject(error);
    return;
  }

  if (info.done) {
    resolve(value);
  } else {
    Promise.resolve(value).then(_next, _throw);
  }
}

function _asyncToGenerator(fn) {
  return function () {
    var self = this,
        args = arguments;
    return new Promise(function (resolve, reject) {
      var gen = fn.apply(self, args);

      function _next(value) {
        asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value);
      }

      function _throw(err) {
        asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err);
      }

      _next(undefined);
    });
  };
}

function createCommonjsModule(fn, module) {
	return module = { exports: {} }, fn(module, module.exports), module.exports;
}

var runtime_1 = createCommonjsModule(function (module) {
/**
 * Copyright (c) 2014-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

var runtime = (function (exports) {

  var Op = Object.prototype;
  var hasOwn = Op.hasOwnProperty;
  var undefined$1; // More compressible than void 0.
  var $Symbol = typeof Symbol === "function" ? Symbol : {};
  var iteratorSymbol = $Symbol.iterator || "@@iterator";
  var asyncIteratorSymbol = $Symbol.asyncIterator || "@@asyncIterator";
  var toStringTagSymbol = $Symbol.toStringTag || "@@toStringTag";

  function define(obj, key, value) {
    Object.defineProperty(obj, key, {
      value: value,
      enumerable: true,
      configurable: true,
      writable: true
    });
    return obj[key];
  }
  try {
    // IE 8 has a broken Object.defineProperty that only works on DOM objects.
    define({}, "");
  } catch (err) {
    define = function(obj, key, value) {
      return obj[key] = value;
    };
  }

  function wrap(innerFn, outerFn, self, tryLocsList) {
    // If outerFn provided and outerFn.prototype is a Generator, then outerFn.prototype instanceof Generator.
    var protoGenerator = outerFn && outerFn.prototype instanceof Generator ? outerFn : Generator;
    var generator = Object.create(protoGenerator.prototype);
    var context = new Context(tryLocsList || []);

    // The ._invoke method unifies the implementations of the .next,
    // .throw, and .return methods.
    generator._invoke = makeInvokeMethod(innerFn, self, context);

    return generator;
  }
  exports.wrap = wrap;

  // Try/catch helper to minimize deoptimizations. Returns a completion
  // record like context.tryEntries[i].completion. This interface could
  // have been (and was previously) designed to take a closure to be
  // invoked without arguments, but in all the cases we care about we
  // already have an existing method we want to call, so there's no need
  // to create a new function object. We can even get away with assuming
  // the method takes exactly one argument, since that happens to be true
  // in every case, so we don't have to touch the arguments object. The
  // only additional allocation required is the completion record, which
  // has a stable shape and so hopefully should be cheap to allocate.
  function tryCatch(fn, obj, arg) {
    try {
      return { type: "normal", arg: fn.call(obj, arg) };
    } catch (err) {
      return { type: "throw", arg: err };
    }
  }

  var GenStateSuspendedStart = "suspendedStart";
  var GenStateSuspendedYield = "suspendedYield";
  var GenStateExecuting = "executing";
  var GenStateCompleted = "completed";

  // Returning this object from the innerFn has the same effect as
  // breaking out of the dispatch switch statement.
  var ContinueSentinel = {};

  // Dummy constructor functions that we use as the .constructor and
  // .constructor.prototype properties for functions that return Generator
  // objects. For full spec compliance, you may wish to configure your
  // minifier not to mangle the names of these two functions.
  function Generator() {}
  function GeneratorFunction() {}
  function GeneratorFunctionPrototype() {}

  // This is a polyfill for %IteratorPrototype% for environments that
  // don't natively support it.
  var IteratorPrototype = {};
  define(IteratorPrototype, iteratorSymbol, function () {
    return this;
  });

  var getProto = Object.getPrototypeOf;
  var NativeIteratorPrototype = getProto && getProto(getProto(values([])));
  if (NativeIteratorPrototype &&
      NativeIteratorPrototype !== Op &&
      hasOwn.call(NativeIteratorPrototype, iteratorSymbol)) {
    // This environment has a native %IteratorPrototype%; use it instead
    // of the polyfill.
    IteratorPrototype = NativeIteratorPrototype;
  }

  var Gp = GeneratorFunctionPrototype.prototype =
    Generator.prototype = Object.create(IteratorPrototype);
  GeneratorFunction.prototype = GeneratorFunctionPrototype;
  define(Gp, "constructor", GeneratorFunctionPrototype);
  define(GeneratorFunctionPrototype, "constructor", GeneratorFunction);
  GeneratorFunction.displayName = define(
    GeneratorFunctionPrototype,
    toStringTagSymbol,
    "GeneratorFunction"
  );

  // Helper for defining the .next, .throw, and .return methods of the
  // Iterator interface in terms of a single ._invoke method.
  function defineIteratorMethods(prototype) {
    ["next", "throw", "return"].forEach(function(method) {
      define(prototype, method, function(arg) {
        return this._invoke(method, arg);
      });
    });
  }

  exports.isGeneratorFunction = function(genFun) {
    var ctor = typeof genFun === "function" && genFun.constructor;
    return ctor
      ? ctor === GeneratorFunction ||
        // For the native GeneratorFunction constructor, the best we can
        // do is to check its .name property.
        (ctor.displayName || ctor.name) === "GeneratorFunction"
      : false;
  };

  exports.mark = function(genFun) {
    if (Object.setPrototypeOf) {
      Object.setPrototypeOf(genFun, GeneratorFunctionPrototype);
    } else {
      genFun.__proto__ = GeneratorFunctionPrototype;
      define(genFun, toStringTagSymbol, "GeneratorFunction");
    }
    genFun.prototype = Object.create(Gp);
    return genFun;
  };

  // Within the body of any async function, `await x` is transformed to
  // `yield regeneratorRuntime.awrap(x)`, so that the runtime can test
  // `hasOwn.call(value, "__await")` to determine if the yielded value is
  // meant to be awaited.
  exports.awrap = function(arg) {
    return { __await: arg };
  };

  function AsyncIterator(generator, PromiseImpl) {
    function invoke(method, arg, resolve, reject) {
      var record = tryCatch(generator[method], generator, arg);
      if (record.type === "throw") {
        reject(record.arg);
      } else {
        var result = record.arg;
        var value = result.value;
        if (value &&
            typeof value === "object" &&
            hasOwn.call(value, "__await")) {
          return PromiseImpl.resolve(value.__await).then(function(value) {
            invoke("next", value, resolve, reject);
          }, function(err) {
            invoke("throw", err, resolve, reject);
          });
        }

        return PromiseImpl.resolve(value).then(function(unwrapped) {
          // When a yielded Promise is resolved, its final value becomes
          // the .value of the Promise<{value,done}> result for the
          // current iteration.
          result.value = unwrapped;
          resolve(result);
        }, function(error) {
          // If a rejected Promise was yielded, throw the rejection back
          // into the async generator function so it can be handled there.
          return invoke("throw", error, resolve, reject);
        });
      }
    }

    var previousPromise;

    function enqueue(method, arg) {
      function callInvokeWithMethodAndArg() {
        return new PromiseImpl(function(resolve, reject) {
          invoke(method, arg, resolve, reject);
        });
      }

      return previousPromise =
        // If enqueue has been called before, then we want to wait until
        // all previous Promises have been resolved before calling invoke,
        // so that results are always delivered in the correct order. If
        // enqueue has not been called before, then it is important to
        // call invoke immediately, without waiting on a callback to fire,
        // so that the async generator function has the opportunity to do
        // any necessary setup in a predictable way. This predictability
        // is why the Promise constructor synchronously invokes its
        // executor callback, and why async functions synchronously
        // execute code before the first await. Since we implement simple
        // async functions in terms of async generators, it is especially
        // important to get this right, even though it requires care.
        previousPromise ? previousPromise.then(
          callInvokeWithMethodAndArg,
          // Avoid propagating failures to Promises returned by later
          // invocations of the iterator.
          callInvokeWithMethodAndArg
        ) : callInvokeWithMethodAndArg();
    }

    // Define the unified helper method that is used to implement .next,
    // .throw, and .return (see defineIteratorMethods).
    this._invoke = enqueue;
  }

  defineIteratorMethods(AsyncIterator.prototype);
  define(AsyncIterator.prototype, asyncIteratorSymbol, function () {
    return this;
  });
  exports.AsyncIterator = AsyncIterator;

  // Note that simple async functions are implemented on top of
  // AsyncIterator objects; they just return a Promise for the value of
  // the final result produced by the iterator.
  exports.async = function(innerFn, outerFn, self, tryLocsList, PromiseImpl) {
    if (PromiseImpl === void 0) PromiseImpl = Promise;

    var iter = new AsyncIterator(
      wrap(innerFn, outerFn, self, tryLocsList),
      PromiseImpl
    );

    return exports.isGeneratorFunction(outerFn)
      ? iter // If outerFn is a generator, return the full iterator.
      : iter.next().then(function(result) {
          return result.done ? result.value : iter.next();
        });
  };

  function makeInvokeMethod(innerFn, self, context) {
    var state = GenStateSuspendedStart;

    return function invoke(method, arg) {
      if (state === GenStateExecuting) {
        throw new Error("Generator is already running");
      }

      if (state === GenStateCompleted) {
        if (method === "throw") {
          throw arg;
        }

        // Be forgiving, per 25.3.3.3.3 of the spec:
        // https://people.mozilla.org/~jorendorff/es6-draft.html#sec-generatorresume
        return doneResult();
      }

      context.method = method;
      context.arg = arg;

      while (true) {
        var delegate = context.delegate;
        if (delegate) {
          var delegateResult = maybeInvokeDelegate(delegate, context);
          if (delegateResult) {
            if (delegateResult === ContinueSentinel) continue;
            return delegateResult;
          }
        }

        if (context.method === "next") {
          // Setting context._sent for legacy support of Babel's
          // function.sent implementation.
          context.sent = context._sent = context.arg;

        } else if (context.method === "throw") {
          if (state === GenStateSuspendedStart) {
            state = GenStateCompleted;
            throw context.arg;
          }

          context.dispatchException(context.arg);

        } else if (context.method === "return") {
          context.abrupt("return", context.arg);
        }

        state = GenStateExecuting;

        var record = tryCatch(innerFn, self, context);
        if (record.type === "normal") {
          // If an exception is thrown from innerFn, we leave state ===
          // GenStateExecuting and loop back for another invocation.
          state = context.done
            ? GenStateCompleted
            : GenStateSuspendedYield;

          if (record.arg === ContinueSentinel) {
            continue;
          }

          return {
            value: record.arg,
            done: context.done
          };

        } else if (record.type === "throw") {
          state = GenStateCompleted;
          // Dispatch the exception by looping back around to the
          // context.dispatchException(context.arg) call above.
          context.method = "throw";
          context.arg = record.arg;
        }
      }
    };
  }

  // Call delegate.iterator[context.method](context.arg) and handle the
  // result, either by returning a { value, done } result from the
  // delegate iterator, or by modifying context.method and context.arg,
  // setting context.delegate to null, and returning the ContinueSentinel.
  function maybeInvokeDelegate(delegate, context) {
    var method = delegate.iterator[context.method];
    if (method === undefined$1) {
      // A .throw or .return when the delegate iterator has no .throw
      // method always terminates the yield* loop.
      context.delegate = null;

      if (context.method === "throw") {
        // Note: ["return"] must be used for ES3 parsing compatibility.
        if (delegate.iterator["return"]) {
          // If the delegate iterator has a return method, give it a
          // chance to clean up.
          context.method = "return";
          context.arg = undefined$1;
          maybeInvokeDelegate(delegate, context);

          if (context.method === "throw") {
            // If maybeInvokeDelegate(context) changed context.method from
            // "return" to "throw", let that override the TypeError below.
            return ContinueSentinel;
          }
        }

        context.method = "throw";
        context.arg = new TypeError(
          "The iterator does not provide a 'throw' method");
      }

      return ContinueSentinel;
    }

    var record = tryCatch(method, delegate.iterator, context.arg);

    if (record.type === "throw") {
      context.method = "throw";
      context.arg = record.arg;
      context.delegate = null;
      return ContinueSentinel;
    }

    var info = record.arg;

    if (! info) {
      context.method = "throw";
      context.arg = new TypeError("iterator result is not an object");
      context.delegate = null;
      return ContinueSentinel;
    }

    if (info.done) {
      // Assign the result of the finished delegate to the temporary
      // variable specified by delegate.resultName (see delegateYield).
      context[delegate.resultName] = info.value;

      // Resume execution at the desired location (see delegateYield).
      context.next = delegate.nextLoc;

      // If context.method was "throw" but the delegate handled the
      // exception, let the outer generator proceed normally. If
      // context.method was "next", forget context.arg since it has been
      // "consumed" by the delegate iterator. If context.method was
      // "return", allow the original .return call to continue in the
      // outer generator.
      if (context.method !== "return") {
        context.method = "next";
        context.arg = undefined$1;
      }

    } else {
      // Re-yield the result returned by the delegate method.
      return info;
    }

    // The delegate iterator is finished, so forget it and continue with
    // the outer generator.
    context.delegate = null;
    return ContinueSentinel;
  }

  // Define Generator.prototype.{next,throw,return} in terms of the
  // unified ._invoke helper method.
  defineIteratorMethods(Gp);

  define(Gp, toStringTagSymbol, "Generator");

  // A Generator should always return itself as the iterator object when the
  // @@iterator function is called on it. Some browsers' implementations of the
  // iterator prototype chain incorrectly implement this, causing the Generator
  // object to not be returned from this call. This ensures that doesn't happen.
  // See https://github.com/facebook/regenerator/issues/274 for more details.
  define(Gp, iteratorSymbol, function() {
    return this;
  });

  define(Gp, "toString", function() {
    return "[object Generator]";
  });

  function pushTryEntry(locs) {
    var entry = { tryLoc: locs[0] };

    if (1 in locs) {
      entry.catchLoc = locs[1];
    }

    if (2 in locs) {
      entry.finallyLoc = locs[2];
      entry.afterLoc = locs[3];
    }

    this.tryEntries.push(entry);
  }

  function resetTryEntry(entry) {
    var record = entry.completion || {};
    record.type = "normal";
    delete record.arg;
    entry.completion = record;
  }

  function Context(tryLocsList) {
    // The root entry object (effectively a try statement without a catch
    // or a finally block) gives us a place to store values thrown from
    // locations where there is no enclosing try statement.
    this.tryEntries = [{ tryLoc: "root" }];
    tryLocsList.forEach(pushTryEntry, this);
    this.reset(true);
  }

  exports.keys = function(object) {
    var keys = [];
    for (var key in object) {
      keys.push(key);
    }
    keys.reverse();

    // Rather than returning an object with a next method, we keep
    // things simple and return the next function itself.
    return function next() {
      while (keys.length) {
        var key = keys.pop();
        if (key in object) {
          next.value = key;
          next.done = false;
          return next;
        }
      }

      // To avoid creating an additional object, we just hang the .value
      // and .done properties off the next function object itself. This
      // also ensures that the minifier will not anonymize the function.
      next.done = true;
      return next;
    };
  };

  function values(iterable) {
    if (iterable) {
      var iteratorMethod = iterable[iteratorSymbol];
      if (iteratorMethod) {
        return iteratorMethod.call(iterable);
      }

      if (typeof iterable.next === "function") {
        return iterable;
      }

      if (!isNaN(iterable.length)) {
        var i = -1, next = function next() {
          while (++i < iterable.length) {
            if (hasOwn.call(iterable, i)) {
              next.value = iterable[i];
              next.done = false;
              return next;
            }
          }

          next.value = undefined$1;
          next.done = true;

          return next;
        };

        return next.next = next;
      }
    }

    // Return an iterator with no values.
    return { next: doneResult };
  }
  exports.values = values;

  function doneResult() {
    return { value: undefined$1, done: true };
  }

  Context.prototype = {
    constructor: Context,

    reset: function(skipTempReset) {
      this.prev = 0;
      this.next = 0;
      // Resetting context._sent for legacy support of Babel's
      // function.sent implementation.
      this.sent = this._sent = undefined$1;
      this.done = false;
      this.delegate = null;

      this.method = "next";
      this.arg = undefined$1;

      this.tryEntries.forEach(resetTryEntry);

      if (!skipTempReset) {
        for (var name in this) {
          // Not sure about the optimal order of these conditions:
          if (name.charAt(0) === "t" &&
              hasOwn.call(this, name) &&
              !isNaN(+name.slice(1))) {
            this[name] = undefined$1;
          }
        }
      }
    },

    stop: function() {
      this.done = true;

      var rootEntry = this.tryEntries[0];
      var rootRecord = rootEntry.completion;
      if (rootRecord.type === "throw") {
        throw rootRecord.arg;
      }

      return this.rval;
    },

    dispatchException: function(exception) {
      if (this.done) {
        throw exception;
      }

      var context = this;
      function handle(loc, caught) {
        record.type = "throw";
        record.arg = exception;
        context.next = loc;

        if (caught) {
          // If the dispatched exception was caught by a catch block,
          // then let that catch block handle the exception normally.
          context.method = "next";
          context.arg = undefined$1;
        }

        return !! caught;
      }

      for (var i = this.tryEntries.length - 1; i >= 0; --i) {
        var entry = this.tryEntries[i];
        var record = entry.completion;

        if (entry.tryLoc === "root") {
          // Exception thrown outside of any try block that could handle
          // it, so set the completion value of the entire function to
          // throw the exception.
          return handle("end");
        }

        if (entry.tryLoc <= this.prev) {
          var hasCatch = hasOwn.call(entry, "catchLoc");
          var hasFinally = hasOwn.call(entry, "finallyLoc");

          if (hasCatch && hasFinally) {
            if (this.prev < entry.catchLoc) {
              return handle(entry.catchLoc, true);
            } else if (this.prev < entry.finallyLoc) {
              return handle(entry.finallyLoc);
            }

          } else if (hasCatch) {
            if (this.prev < entry.catchLoc) {
              return handle(entry.catchLoc, true);
            }

          } else if (hasFinally) {
            if (this.prev < entry.finallyLoc) {
              return handle(entry.finallyLoc);
            }

          } else {
            throw new Error("try statement without catch or finally");
          }
        }
      }
    },

    abrupt: function(type, arg) {
      for (var i = this.tryEntries.length - 1; i >= 0; --i) {
        var entry = this.tryEntries[i];
        if (entry.tryLoc <= this.prev &&
            hasOwn.call(entry, "finallyLoc") &&
            this.prev < entry.finallyLoc) {
          var finallyEntry = entry;
          break;
        }
      }

      if (finallyEntry &&
          (type === "break" ||
           type === "continue") &&
          finallyEntry.tryLoc <= arg &&
          arg <= finallyEntry.finallyLoc) {
        // Ignore the finally entry if control is not jumping to a
        // location outside the try/catch block.
        finallyEntry = null;
      }

      var record = finallyEntry ? finallyEntry.completion : {};
      record.type = type;
      record.arg = arg;

      if (finallyEntry) {
        this.method = "next";
        this.next = finallyEntry.finallyLoc;
        return ContinueSentinel;
      }

      return this.complete(record);
    },

    complete: function(record, afterLoc) {
      if (record.type === "throw") {
        throw record.arg;
      }

      if (record.type === "break" ||
          record.type === "continue") {
        this.next = record.arg;
      } else if (record.type === "return") {
        this.rval = this.arg = record.arg;
        this.method = "return";
        this.next = "end";
      } else if (record.type === "normal" && afterLoc) {
        this.next = afterLoc;
      }

      return ContinueSentinel;
    },

    finish: function(finallyLoc) {
      for (var i = this.tryEntries.length - 1; i >= 0; --i) {
        var entry = this.tryEntries[i];
        if (entry.finallyLoc === finallyLoc) {
          this.complete(entry.completion, entry.afterLoc);
          resetTryEntry(entry);
          return ContinueSentinel;
        }
      }
    },

    "catch": function(tryLoc) {
      for (var i = this.tryEntries.length - 1; i >= 0; --i) {
        var entry = this.tryEntries[i];
        if (entry.tryLoc === tryLoc) {
          var record = entry.completion;
          if (record.type === "throw") {
            var thrown = record.arg;
            resetTryEntry(entry);
          }
          return thrown;
        }
      }

      // The context.catch method must only be called with a location
      // argument that corresponds to a known catch block.
      throw new Error("illegal catch attempt");
    },

    delegateYield: function(iterable, resultName, nextLoc) {
      this.delegate = {
        iterator: values(iterable),
        resultName: resultName,
        nextLoc: nextLoc
      };

      if (this.method === "next") {
        // Deliberately forget the last sent value so that we don't
        // accidentally pass it on to the delegate.
        this.arg = undefined$1;
      }

      return ContinueSentinel;
    }
  };

  // Regardless of whether this script is executing as a CommonJS module
  // or not, return the runtime object so that we can declare the variable
  // regeneratorRuntime in the outer scope, which allows this module to be
  // injected easily by `bin/regenerator --include-runtime script.js`.
  return exports;

}(
  // If this script is executing as a CommonJS module, use module.exports
  // as the regeneratorRuntime namespace. Otherwise create a new empty
  // object. Either way, the resulting object will be used to initialize
  // the regeneratorRuntime variable at the top of this file.
   module.exports 
));

try {
  regeneratorRuntime = runtime;
} catch (accidentalStrictMode) {
  // This module should not be running in strict mode, so the above
  // assignment should always work unless something is misconfigured. Just
  // in case runtime.js accidentally runs in strict mode, in modern engines
  // we can explicitly access globalThis. In older engines we can escape
  // strict mode using a global Function call. This could conceivably fail
  // if a Content Security Policy forbids using Function, but in that case
  // the proper solution is to fix the accidental strict mode problem. If
  // you've misconfigured your bundler to force strict mode and applied a
  // CSP to forbid Function, and you're not willing to fix either of those
  // problems, please detail your unique predicament in a GitHub issue.
  if (typeof globalThis === "object") {
    globalThis.regeneratorRuntime = runtime;
  } else {
    Function("r", "regeneratorRuntime = r")(runtime);
  }
}
});

var users = [{
  id: 'u_1',
  name: 'Tyler',
  profilePictureUrl: 'https://res.cloudinary.com/uidotdev/image/twitter_name/tylermcginnis'
}, {
  id: 'u_2',
  name: 'Bono',
  profilePictureUrl: 'https://res.cloudinary.com/uidotdev/image/twitter_name/u2'
}, {
  id: 'u_3',
  name: 'Tanner',
  profilePictureUrl: 'https://res.cloudinary.com/uidotdev/image/twitter_name/tannerlinsley'
}, {
  id: 'u_4',
  name: 'Alex',
  profilePictureUrl: 'https://res.cloudinary.com/uidotdev/image/twitter_name/ralex1993'
}];
var labels = [{
  id: 'bug',
  color: 'red',
  name: 'bug'
}, {
  id: 'feature',
  color: 'blue',
  name: 'feature'
}, {
  id: 'enhancement',
  color: 'cyan',
  name: 'enhancement'
}, {
  id: 'question',
  color: 'orange',
  name: 'question'
}, {
  id: 'help',
  color: 'lime',
  name: 'help wanted'
}, {
  id: 'wontfix',
  color: 'white',
  name: 'wontfix'
}, {
  id: 'duplicate',
  color: 'rebeccapurple',
  name: 'duplicate'
}];
var part1 = ['Dependencies', 'The App', 'Windows', 'macOS', 'Styling', 'Button', 'Target', 'Input', 'Field', 'JavaScript', 'React', 'JQuery', 'My Cookies'];
var part2 = ['is having a problem', 'seems to struggle', 'throws an error', 'makes my computer run slow', 'causes the processor to heat up', 'looks weird', "cannot read property 'length' of undefined", 'is not working', 'is not responding', 'is not working properly', 'is not working as expected', 'is crashing', "won't run right", "is actually working fine. I just wanted to let you know you're great"];
var part3 = ['when I rage click it', 'on Tuesdays', 'every time I wear my green shirt', "when I'm on a plane", "when I'm on a train", "when I'm on a boat", "when I'm on a bike", 'right now', 'all the time', 'on weekends', "when I'm with Taylor Swift", 'whenever I try to demo it'];
var templateIssueComments = ["I'm on it!", "I'm not sure what the problem is.", "I'm working on it.", "I'm not sure how to fix it.", "I'm not sure if I can reproduce the problem.", 'This is a really big deal for me.', 'Has there been any progress on this?', 'What is the status of this issue?', 'Never mind, I figured out how to fix this', 'Can you send me a little bit more information about the problem.', "I've reproduced this issue. Working on a fix now.", "I'm on it. I'll get back to you when I'm done.", 'It would seem this is caused by user error.', 'Whoops, I just dropped the production database. Hang on...'];
var allStatus = ['backlog', 'todo', 'inProgress', 'done', 'cancelled'];
var issueComments = /*#__PURE__*/Array.from({
  length: 500
}, function (_, j) {
  return {
    id: "c_" + j,
    createdDate: new Date(Date.now() - Math.floor(Math.random() * 10000000)),
    createdBy: users[Math.floor(Math.random() * users.length)].id,
    comment: templateIssueComments[Math.floor(Math.random() * templateIssueComments.length)]
  };
}).sort(function (a, b) {
  if (a.createdDate < b.createdDate) {
    return -1;
  }

  if (a.createdDate > b.createdDate) {
    return 1;
  }

  return 0;
});
var issues = /*#__PURE__*/Array.from({
  length: 1000
}, function (_, i) {
  var isCompleted = Math.random() > 0.9;
  var title = part1[Math.floor(Math.random() * part1.length)] + " " + part2[Math.floor(Math.random() * part2.length)] + " " + part3[Math.floor(Math.random() * part3.length)];
  return {
    id: "i_" + i,
    title: title,
    labels: [labels[Math.floor(Math.random() * labels.length)].id],
    comments: Array.from({
      length: Math.round(Math.random() * 500)
    }).map(function () {
      return "c_" + Math.round(Math.random() * 500);
    }).filter(function (a, i, arr) {
      return arr.indexOf(a) === i;
    }),
    number: i + 1,
    status: isCompleted ? 'done' : allStatus.filter(function (f) {
      return f !== 'done';
    })[Math.floor(Math.random() * allStatus.length)],
    createdDate: new Date(Date.now() - Math.floor(Math.random() * 10000000)),
    createdBy: users[Math.floor(Math.random() * users.length)].id,
    assignee: Math.random() > 0.5 ? users[Math.floor(Math.random() * users.length)].id : null,
    dueDate: Math.random() > 0.5 ? new Date(Date.now() + Math.floor(Math.random() * 10000000)) : null,
    completedDate: isCompleted ? new Date(Date.now() + Math.floor(Math.random() * 10000000)) : null
  };
});

var makeUrl = function makeUrl(path) {
  return "" + (typeof window === 'undefined' ? 'http://localhost:8000' : '') + path;
};

var handleErrorDelay = /*#__PURE__*/function () {
  var _ref = /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee(req) {
    return runtime_1.wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            if (!req.headers.get('x-delay')) {
              _context.next = 5;
              break;
            }

            _context.next = 3;
            return new Promise(function (resolve) {
              return setTimeout(resolve, Number(req.headers.get('x-delay')) || 0);
            });

          case 3:
            _context.next = 7;
            break;

          case 5:
            _context.next = 7;
            return new Promise(function (resolve) {
              return setTimeout(resolve, Math.random() * 500 + 500);
            });

          case 7:
            if (!req.headers.get('x-error')) {
              _context.next = 10;
              break;
            }

            if (!(Math.random() > 0.5)) {
              _context.next = 10;
              break;
            }

            throw new Error();

          case 10:
          case "end":
            return _context.stop();
        }
      }
    }, _callee);
  }));

  return function handleErrorDelay(_x) {
    return _ref.apply(this, arguments);
  };
}();

var handlers = [/*#__PURE__*/msw.rest.get( /*#__PURE__*/makeUrl('/api/status'), /*#__PURE__*/function () {
  var _ref2 = /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee2(_req, res, ctx) {
    return runtime_1.wrap(function _callee2$(_context2) {
      while (1) {
        switch (_context2.prev = _context2.next) {
          case 0:
            return _context2.abrupt("return", res(ctx.status(200), ctx.json({
              status: 'ok'
            })));

          case 1:
          case "end":
            return _context2.stop();
        }
      }
    }, _callee2);
  }));

  return function (_x2, _x3, _x4) {
    return _ref2.apply(this, arguments);
  };
}()), /*#__PURE__*/msw.rest.get( /*#__PURE__*/makeUrl('/api/issues'), /*#__PURE__*/function () {
  var _ref3 = /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee3(req, res, ctx) {
    var query, page, perPage, statusFilter, labelQuery, order, filteredIssues, sortedIssues, pagedIssues;
    return runtime_1.wrap(function _callee3$(_context3) {
      while (1) {
        switch (_context3.prev = _context3.next) {
          case 0:
            _context3.prev = 0;
            _context3.next = 3;
            return handleErrorDelay(req);

          case 3:
            _context3.next = 8;
            break;

          case 5:
            _context3.prev = 5;
            _context3.t0 = _context3["catch"](0);
            return _context3.abrupt("return", res(ctx.status(500), ctx.json({
              error: 'Error in request'
            })));

          case 8:
            query = req.url.searchParams;
            page = Number(query.get('page')) || 1;
            perPage = Number(query.get('limit')) || 10;
            statusFilter = query.get('status');
            labelQuery = query.getAll('labels[]');
            order = query.get('order') || 'desc';
            filteredIssues = issues.filter(function (issue) {
              if (statusFilter) {
                if (issue.status !== statusFilter) return false;
              } else {
                if (issue.status === 'done' || issue.status === 'cancelled') return false;
              }

              if (labelQuery.length > 0) {
                if (!labelQuery.some(function (label) {
                  var dbLabel = labels.find(function (l) {
                    return l.name === label;
                  });
                  if (!dbLabel) return false;
                  return issue.labels.find(function (l) {
                    return l === dbLabel.id;
                  });
                })) {
                  return false;
                }
              }

              return true;
            });
            sortedIssues = filteredIssues.sort(function (a, b) {
              if (order === 'asc') {
                if (a.number < b.number) return -1;
                if (a.number > b.number) return 1;
                return 0;
              } else {
                if (a.number < b.number) return 1;
                if (a.number > b.number) return -1;
                return 0;
              }
            });
            pagedIssues = sortedIssues.slice((page - 1) * perPage, page * perPage);
            return _context3.abrupt("return", res(ctx.status(200), ctx.json(pagedIssues)));

          case 18:
          case "end":
            return _context3.stop();
        }
      }
    }, _callee3, null, [[0, 5]]);
  }));

  return function (_x5, _x6, _x7) {
    return _ref3.apply(this, arguments);
  };
}()), /*#__PURE__*/msw.rest.get( /*#__PURE__*/makeUrl('/api/issues/:number'), /*#__PURE__*/function () {
  var _ref4 = /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee4(req, res, ctx) {
    var number, issue;
    return runtime_1.wrap(function _callee4$(_context4) {
      while (1) {
        switch (_context4.prev = _context4.next) {
          case 0:
            _context4.prev = 0;
            _context4.next = 3;
            return handleErrorDelay(req);

          case 3:
            _context4.next = 8;
            break;

          case 5:
            _context4.prev = 5;
            _context4.t0 = _context4["catch"](0);
            return _context4.abrupt("return", res(ctx.status(500), ctx.json({
              error: 'Error in request'
            })));

          case 8:
            number = Number(req.params.number);
            issue = issues.find(function (issue) {
              return issue.number === number;
            });

            if (issue) {
              _context4.next = 12;
              break;
            }

            return _context4.abrupt("return", res(ctx.status(404), ctx.json({
              message: 'Not found'
            })));

          case 12:
            return _context4.abrupt("return", res(ctx.status(200), ctx.json(issue)));

          case 13:
          case "end":
            return _context4.stop();
        }
      }
    }, _callee4, null, [[0, 5]]);
  }));

  return function (_x8, _x9, _x10) {
    return _ref4.apply(this, arguments);
  };
}()), /*#__PURE__*/msw.rest.get( /*#__PURE__*/makeUrl('/api/issues/:number/comments'), /*#__PURE__*/function () {
  var _ref5 = /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee5(req, res, ctx) {
    var number, issue, query, page, perPage, order, filteredComments, sortedIssues, pagedComments;
    return runtime_1.wrap(function _callee5$(_context5) {
      while (1) {
        switch (_context5.prev = _context5.next) {
          case 0:
            _context5.prev = 0;
            _context5.next = 3;
            return handleErrorDelay(req);

          case 3:
            _context5.next = 8;
            break;

          case 5:
            _context5.prev = 5;
            _context5.t0 = _context5["catch"](0);
            return _context5.abrupt("return", res(ctx.status(500), ctx.json({
              error: 'Error in request'
            })));

          case 8:
            number = Number(req.params.number);
            issue = issues.find(function (issue) {
              return issue.number === number;
            });

            if (issue) {
              _context5.next = 12;
              break;
            }

            return _context5.abrupt("return", res(ctx.status(404), ctx.json({
              message: 'Not found'
            })));

          case 12:
            query = req.url.searchParams;
            page = Number(query.get('page')) || 1;
            perPage = Number(query.get('limit')) || 10;
            order = query.get('order') || 'desc';
            filteredComments = issue.comments.map(function (id) {
              return issueComments.find(function (comment) {
                return comment.id === id;
              });
            }).filter(Boolean);
            sortedIssues = filteredComments.sort(function (a, b) {
              if (order === 'asc') {
                if (a.createdDate < b.createdDate) return -1;
                if (a.createdDate > b.createdDate) return 1;
                return 0;
              } else {
                if (a.createdDate < b.createdDate) return 1;
                if (a.createdDate > b.createdDate) return -1;
                return 0;
              }
            });
            pagedComments = sortedIssues.slice((page - 1) * perPage, page * perPage);
            return _context5.abrupt("return", res(ctx.status(200), ctx.json(pagedComments)));

          case 20:
          case "end":
            return _context5.stop();
        }
      }
    }, _callee5, null, [[0, 5]]);
  }));

  return function (_x11, _x12, _x13) {
    return _ref5.apply(this, arguments);
  };
}()), /*#__PURE__*/msw.rest.post( /*#__PURE__*/makeUrl('/api/issues/:number/comments'), /*#__PURE__*/function () {
  var _ref6 = /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee6(req, res, ctx) {
    var number, issue, body, createdBy, comment;
    return runtime_1.wrap(function _callee6$(_context6) {
      while (1) {
        switch (_context6.prev = _context6.next) {
          case 0:
            _context6.prev = 0;
            _context6.next = 3;
            return handleErrorDelay(req);

          case 3:
            _context6.next = 8;
            break;

          case 5:
            _context6.prev = 5;
            _context6.t0 = _context6["catch"](0);
            return _context6.abrupt("return", res(ctx.status(500), ctx.json({
              error: 'Error in request'
            })));

          case 8:
            number = Number(req.params.number);
            issue = issues.find(function (issue) {
              return issue.number === number;
            });

            if (issue) {
              _context6.next = 12;
              break;
            }

            return _context6.abrupt("return", res(ctx.status(404), ctx.json({
              message: 'Not found'
            })));

          case 12:
            body = JSON.parse(req.body);
            createdBy = users.find(function (user) {
              return user.id === body.createdBy_id;
            });

            if (createdBy) {
              _context6.next = 16;
              break;
            }

            return _context6.abrupt("return", res(ctx.status(400), ctx.json({
              message: 'Not found'
            })));

          case 16:
            comment = {
              id: "c_" + issue.id + "_" + (issue.comments.length + 1),
              createdDate: new Date(),
              createdBy: createdBy.id,
              comment: body.comment
            };
            issue.comments.push();
            return _context6.abrupt("return", res(ctx.status(201), ctx.json(comment)));

          case 19:
          case "end":
            return _context6.stop();
        }
      }
    }, _callee6, null, [[0, 5]]);
  }));

  return function (_x14, _x15, _x16) {
    return _ref6.apply(this, arguments);
  };
}()), /*#__PURE__*/msw.rest.put( /*#__PURE__*/makeUrl('/api/issues/:number'), /*#__PURE__*/function () {
  var _ref7 = /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee7(req, res, ctx) {
    var number, issue, body, _users$find;

    return runtime_1.wrap(function _callee7$(_context7) {
      while (1) {
        switch (_context7.prev = _context7.next) {
          case 0:
            _context7.prev = 0;
            _context7.next = 3;
            return handleErrorDelay(req);

          case 3:
            _context7.next = 8;
            break;

          case 5:
            _context7.prev = 5;
            _context7.t0 = _context7["catch"](0);
            return _context7.abrupt("return", res(ctx.status(500), ctx.json({
              error: 'Error in request'
            })));

          case 8:
            number = Number(req.params.number);
            issue = issues.find(function (issue) {
              return issue.number === number;
            });

            if (issue) {
              _context7.next = 12;
              break;
            }

            return _context7.abrupt("return", res(ctx.status(404), ctx.json({
              message: 'Not found'
            })));

          case 12:
            body = {};
            if (typeof req.body === 'string') body = JSON.parse(req.body);
            if (typeof req.body === 'object') body = req.body;

            if (body.title) {
              issue.title = body.title;
            }

            if (body.status) {
              issue.status = body.status;
            }

            if (body.labels) {
              issue.labels = body.labels.map(function (l) {
                var _labels$find;

                return (_labels$find = labels.find(function (lbl) {
                  return lbl.id === l;
                })) == null ? void 0 : _labels$find.id;
              }).filter(Boolean);
            }

            if (body.dueDate) {
              issue.dueDate = body.dueDate;
            }

            if (body.assignee) {
              issue.assignee = ((_users$find = users.find(function (user) {
                return user.id === body.assignee;
              })) == null ? void 0 : _users$find.id) || null;
            }

            return _context7.abrupt("return", res(ctx.status(200), ctx.json(issue)));

          case 21:
          case "end":
            return _context7.stop();
        }
      }
    }, _callee7, null, [[0, 5]]);
  }));

  return function (_x17, _x18, _x19) {
    return _ref7.apply(this, arguments);
  };
}()), /*#__PURE__*/msw.rest.post( /*#__PURE__*/makeUrl('/api/issues/:number/complete'), /*#__PURE__*/function () {
  var _ref8 = /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee8(req, res, ctx) {
    var number, issue;
    return runtime_1.wrap(function _callee8$(_context8) {
      while (1) {
        switch (_context8.prev = _context8.next) {
          case 0:
            _context8.prev = 0;
            _context8.next = 3;
            return handleErrorDelay(req);

          case 3:
            _context8.next = 8;
            break;

          case 5:
            _context8.prev = 5;
            _context8.t0 = _context8["catch"](0);
            return _context8.abrupt("return", res(ctx.status(500), ctx.json({
              error: 'Error in request'
            })));

          case 8:
            number = Number(req.params.number);
            issue = issues.find(function (issue) {
              return issue.number === number;
            });

            if (issue) {
              _context8.next = 12;
              break;
            }

            return _context8.abrupt("return", res(ctx.status(404), ctx.json({
              message: 'Not found'
            })));

          case 12:
            issue.createdDate = new Date();
            issue.status = 'done';
            return _context8.abrupt("return", res(ctx.status(200), ctx.json(issue)));

          case 15:
          case "end":
            return _context8.stop();
        }
      }
    }, _callee8, null, [[0, 5]]);
  }));

  return function (_x20, _x21, _x22) {
    return _ref8.apply(this, arguments);
  };
}()), /*#__PURE__*/msw.rest.post( /*#__PURE__*/makeUrl('/api/issues'), /*#__PURE__*/function () {
  var _ref9 = /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee9(req, res, ctx) {
    var body, number, issueComment, issue;
    return runtime_1.wrap(function _callee9$(_context9) {
      while (1) {
        switch (_context9.prev = _context9.next) {
          case 0:
            _context9.prev = 0;
            _context9.next = 3;
            return handleErrorDelay(req);

          case 3:
            _context9.next = 8;
            break;

          case 5:
            _context9.prev = 5;
            _context9.t0 = _context9["catch"](0);
            return _context9.abrupt("return", res(ctx.status(500), ctx.json({
              error: 'Error in request'
            })));

          case 8:
            body = {};
            if (typeof req.body === 'string') body = JSON.parse(req.body);
            if (typeof req.body === 'object') body = req.body;
            number = issues.length + 1;
            issueComment = {
              issueId: "i_" + number,
              id: "c_" + issueComments.length,
              createdDate: new Date(),
              createdBy: users[Math.floor(Math.random() * users.length)].id,
              comment: body.comment
            };
            issueComments.push(issueComment);
            issue = {
              id: "i_" + number,
              number: number,
              title: body.title,
              status: 'backlog',
              comments: [issueComment.id],
              createdDate: new Date(),
              createdBy: users[Math.floor(Math.random() * users.length)].id,
              dueDate: null,
              completedDate: null,
              assignee: null,
              labels: []
            };
            issues.push(issue);
            return _context9.abrupt("return", res(ctx.status(201), ctx.json(issue)));

          case 17:
          case "end":
            return _context9.stop();
        }
      }
    }, _callee9, null, [[0, 5]]);
  }));

  return function (_x23, _x24, _x25) {
    return _ref9.apply(this, arguments);
  };
}()), /*#__PURE__*/msw.rest.get( /*#__PURE__*/makeUrl('/api/labels'), /*#__PURE__*/function () {
  var _ref10 = /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee10(req, res, ctx) {
    return runtime_1.wrap(function _callee10$(_context10) {
      while (1) {
        switch (_context10.prev = _context10.next) {
          case 0:
            _context10.prev = 0;
            _context10.next = 3;
            return handleErrorDelay(req);

          case 3:
            _context10.next = 8;
            break;

          case 5:
            _context10.prev = 5;
            _context10.t0 = _context10["catch"](0);
            return _context10.abrupt("return", res(ctx.status(500), ctx.json({
              error: 'Error in request'
            })));

          case 8:
            return _context10.abrupt("return", res(ctx.status(200), ctx.json(labels)));

          case 9:
          case "end":
            return _context10.stop();
        }
      }
    }, _callee10, null, [[0, 5]]);
  }));

  return function (_x26, _x27, _x28) {
    return _ref10.apply(this, arguments);
  };
}()), /*#__PURE__*/msw.rest.get( /*#__PURE__*/makeUrl('/api/labels/:labelId'), /*#__PURE__*/function () {
  var _ref11 = /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee11(req, res, ctx) {
    var labelId, label;
    return runtime_1.wrap(function _callee11$(_context11) {
      while (1) {
        switch (_context11.prev = _context11.next) {
          case 0:
            _context11.prev = 0;
            _context11.next = 3;
            return handleErrorDelay(req);

          case 3:
            _context11.next = 8;
            break;

          case 5:
            _context11.prev = 5;
            _context11.t0 = _context11["catch"](0);
            return _context11.abrupt("return", res(ctx.status(500), ctx.json({
              error: 'Error in request'
            })));

          case 8:
            labelId = req.params.labelId;
            label = labels.find(function (l) {
              return l.name === labelId;
            });

            if (label) {
              _context11.next = 12;
              break;
            }

            return _context11.abrupt("return", res(ctx.status(404), ctx.json({
              message: 'Label not found'
            })));

          case 12:
            return _context11.abrupt("return", res(ctx.status(200), ctx.json(label)));

          case 13:
          case "end":
            return _context11.stop();
        }
      }
    }, _callee11, null, [[0, 5]]);
  }));

  return function (_x29, _x30, _x31) {
    return _ref11.apply(this, arguments);
  };
}()), /*#__PURE__*/msw.rest.post( /*#__PURE__*/makeUrl('/api/labels'), /*#__PURE__*/function () {
  var _ref12 = /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee12(req, res, ctx) {
    var body, label;
    return runtime_1.wrap(function _callee12$(_context12) {
      while (1) {
        switch (_context12.prev = _context12.next) {
          case 0:
            _context12.prev = 0;
            _context12.next = 3;
            return handleErrorDelay(req);

          case 3:
            _context12.next = 8;
            break;

          case 5:
            _context12.prev = 5;
            _context12.t0 = _context12["catch"](0);
            return _context12.abrupt("return", res(ctx.status(500), ctx.json({
              error: 'Error in request'
            })));

          case 8:
            body = {};
            if (typeof req.body === 'string') body = JSON.parse(req.body);
            if (typeof req.body === 'object') body = req.body;

            if (body.name) {
              _context12.next = 13;
              break;
            }

            return _context12.abrupt("return", res(ctx.status(400), ctx.json({
              message: 'No name'
            })));

          case 13:
            if (!labels.find(function (l) {
              return l.name === body.name;
            })) {
              _context12.next = 15;
              break;
            }

            return _context12.abrupt("return", res(ctx.status(400), ctx.json({
              message: 'Label already exists'
            })));

          case 15:
            label = {
              id: body.name,
              name: body.name,
              color: body.color || 'red'
            };
            labels.push(label);
            return _context12.abrupt("return", res(ctx.status(200), ctx.json(label)));

          case 18:
          case "end":
            return _context12.stop();
        }
      }
    }, _callee12, null, [[0, 5]]);
  }));

  return function (_x32, _x33, _x34) {
    return _ref12.apply(this, arguments);
  };
}()), /*#__PURE__*/msw.rest.put( /*#__PURE__*/makeUrl('/api/labels/:labelId'), /*#__PURE__*/function () {
  var _ref13 = /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee13(req, res, ctx) {
    var labelId, label, body;
    return runtime_1.wrap(function _callee13$(_context13) {
      while (1) {
        switch (_context13.prev = _context13.next) {
          case 0:
            _context13.prev = 0;
            _context13.next = 3;
            return handleErrorDelay(req);

          case 3:
            _context13.next = 8;
            break;

          case 5:
            _context13.prev = 5;
            _context13.t0 = _context13["catch"](0);
            return _context13.abrupt("return", res(ctx.status(500), ctx.json({
              error: 'Error in request'
            })));

          case 8:
            labelId = req.params.labelId;
            label = labels.find(function (l) {
              return l.name === labelId;
            });

            if (label) {
              _context13.next = 12;
              break;
            }

            return _context13.abrupt("return", res(ctx.status(404), ctx.json({
              message: 'Label not found'
            })));

          case 12:
            body = {};
            if (typeof req.body === 'string') body = JSON.parse(req.body);
            if (typeof req.body === 'object') body = req.body;

            if (body.name) {
              _context13.next = 17;
              break;
            }

            return _context13.abrupt("return", res(ctx.status(400), ctx.json({
              message: 'No name'
            })));

          case 17:
            if (body.name) {
              label.name = body.name;
            }

            if (body.color) {
              label.color = body.color;
            }

            return _context13.abrupt("return", res(ctx.status(200), ctx.json(label)));

          case 20:
          case "end":
            return _context13.stop();
        }
      }
    }, _callee13, null, [[0, 5]]);
  }));

  return function (_x35, _x36, _x37) {
    return _ref13.apply(this, arguments);
  };
}()), /*#__PURE__*/msw.rest["delete"]( /*#__PURE__*/makeUrl('/api/labels/:labelId'), /*#__PURE__*/function () {
  var _ref14 = /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee14(req, res, ctx) {
    var labelId, label;
    return runtime_1.wrap(function _callee14$(_context14) {
      while (1) {
        switch (_context14.prev = _context14.next) {
          case 0:
            _context14.prev = 0;
            _context14.next = 3;
            return handleErrorDelay(req);

          case 3:
            _context14.next = 8;
            break;

          case 5:
            _context14.prev = 5;
            _context14.t0 = _context14["catch"](0);
            return _context14.abrupt("return", res(ctx.status(500), ctx.json({
              error: 'Error in request'
            })));

          case 8:
            labelId = req.params.labelId;
            label = labels.find(function (l) {
              return l.name === labelId;
            });

            if (label) {
              _context14.next = 12;
              break;
            }

            return _context14.abrupt("return", res(ctx.status(404), ctx.json({
              message: 'Label not found'
            })));

          case 12:
            labels.splice(labels.indexOf(label), 1);
            return _context14.abrupt("return", res(ctx.status(200), ctx.json(labels)));

          case 14:
          case "end":
            return _context14.stop();
        }
      }
    }, _callee14, null, [[0, 5]]);
  }));

  return function (_x38, _x39, _x40) {
    return _ref14.apply(this, arguments);
  };
}()), /*#__PURE__*/msw.rest.get( /*#__PURE__*/makeUrl('/api/users'), /*#__PURE__*/function () {
  var _ref15 = /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee15(req, res, ctx) {
    return runtime_1.wrap(function _callee15$(_context15) {
      while (1) {
        switch (_context15.prev = _context15.next) {
          case 0:
            _context15.prev = 0;
            _context15.next = 3;
            return handleErrorDelay(req);

          case 3:
            _context15.next = 8;
            break;

          case 5:
            _context15.prev = 5;
            _context15.t0 = _context15["catch"](0);
            return _context15.abrupt("return", res(ctx.status(500), ctx.json({
              error: 'Error in request'
            })));

          case 8:
            return _context15.abrupt("return", res(ctx.status(200), ctx.json(users)));

          case 9:
          case "end":
            return _context15.stop();
        }
      }
    }, _callee15, null, [[0, 5]]);
  }));

  return function (_x41, _x42, _x43) {
    return _ref15.apply(this, arguments);
  };
}()), /*#__PURE__*/msw.rest.get( /*#__PURE__*/makeUrl('/api/users/:userId'), /*#__PURE__*/function () {
  var _ref16 = /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee16(req, res, ctx) {
    var userId, user;
    return runtime_1.wrap(function _callee16$(_context16) {
      while (1) {
        switch (_context16.prev = _context16.next) {
          case 0:
            _context16.prev = 0;
            _context16.next = 3;
            return handleErrorDelay(req);

          case 3:
            _context16.next = 8;
            break;

          case 5:
            _context16.prev = 5;
            _context16.t0 = _context16["catch"](0);
            return _context16.abrupt("return", res(ctx.status(500), ctx.json({
              error: 'Error in request'
            })));

          case 8:
            userId = req.params.userId;
            user = users.find(function (l) {
              return l.id === userId;
            });

            if (user) {
              _context16.next = 12;
              break;
            }

            return _context16.abrupt("return", res(ctx.status(404), ctx.json({
              message: 'User not found'
            })));

          case 12:
            return _context16.abrupt("return", res(ctx.status(200), ctx.json(user)));

          case 13:
          case "end":
            return _context16.stop();
        }
      }
    }, _callee16, null, [[0, 5]]);
  }));

  return function (_x44, _x45, _x46) {
    return _ref16.apply(this, arguments);
  };
}()), /*#__PURE__*/msw.rest.get( /*#__PURE__*/makeUrl('/api/search/issues'), /*#__PURE__*/function () {
  var _ref17 = /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee17(req, res, ctx) {
    var query, filteredList;
    return runtime_1.wrap(function _callee17$(_context17) {
      while (1) {
        switch (_context17.prev = _context17.next) {
          case 0:
            _context17.prev = 0;
            _context17.next = 3;
            return handleErrorDelay(req);

          case 3:
            _context17.next = 8;
            break;

          case 5:
            _context17.prev = 5;
            _context17.t0 = _context17["catch"](0);
            return _context17.abrupt("return", res(ctx.status(500), ctx.json({
              error: 'Error in request'
            })));

          case 8:
            query = req.url.searchParams.get('q') || '';

            if (query) {
              _context17.next = 11;
              break;
            }

            return _context17.abrupt("return", res(ctx.status(401), ctx.json({
              message: 'Search query is required'
            })));

          case 11:
            filteredList = issues.filter(function (issue) {
              return issue.title.includes(query);
            });
            return _context17.abrupt("return", res(ctx.status(200), ctx.json({
              count: filteredList.length,
              items: filteredList
            })));

          case 13:
          case "end":
            return _context17.stop();
        }
      }
    }, _callee17, null, [[0, 5]]);
  }));

  return function (_x47, _x48, _x49) {
    return _ref17.apply(this, arguments);
  };
}()), /*#__PURE__*/msw.rest.get( /*#__PURE__*/makeUrl('/api/search/labels'), /*#__PURE__*/function () {
  var _ref18 = /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee18(req, res, ctx) {
    var query, filteredList;
    return runtime_1.wrap(function _callee18$(_context18) {
      while (1) {
        switch (_context18.prev = _context18.next) {
          case 0:
            _context18.prev = 0;
            _context18.next = 3;
            return handleErrorDelay(req);

          case 3:
            _context18.next = 8;
            break;

          case 5:
            _context18.prev = 5;
            _context18.t0 = _context18["catch"](0);
            return _context18.abrupt("return", res(ctx.status(500), ctx.json({
              error: 'Error in request'
            })));

          case 8:
            query = req.url.searchParams.get('q') || '';

            if (query) {
              _context18.next = 11;
              break;
            }

            return _context18.abrupt("return", res(ctx.status(401), ctx.json({
              message: 'Search query is required'
            })));

          case 11:
            filteredList = labels.filter(function (label) {
              return label.name.includes(query);
            });
            return _context18.abrupt("return", res(ctx.status(200), ctx.json({
              count: filteredList.length,
              items: filteredList
            })));

          case 13:
          case "end":
            return _context18.stop();
        }
      }
    }, _callee18, null, [[0, 5]]);
  }));

  return function (_x50, _x51, _x52) {
    return _ref18.apply(this, arguments);
  };
}()), /*#__PURE__*/msw.rest.get( /*#__PURE__*/makeUrl('/api/search/comments'), /*#__PURE__*/function () {
  var _ref19 = /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee19(req, res, ctx) {
    var query, filteredList;
    return runtime_1.wrap(function _callee19$(_context19) {
      while (1) {
        switch (_context19.prev = _context19.next) {
          case 0:
            _context19.prev = 0;
            _context19.next = 3;
            return handleErrorDelay(req);

          case 3:
            _context19.next = 8;
            break;

          case 5:
            _context19.prev = 5;
            _context19.t0 = _context19["catch"](0);
            return _context19.abrupt("return", res(ctx.status(500), ctx.json({
              error: 'Error in request'
            })));

          case 8:
            query = req.url.searchParams.get('q') || '';

            if (query) {
              _context19.next = 11;
              break;
            }

            return _context19.abrupt("return", res(ctx.status(401), ctx.json({
              message: 'Search query is required'
            })));

          case 11:
            filteredList = issueComments.filter(function (comment) {
              return comment.comment.includes(query);
            });
            return _context19.abrupt("return", res(ctx.status(200), ctx.json({
              count: filteredList.length,
              items: filteredList
            })));

          case 13:
          case "end":
            return _context19.stop();
        }
      }
    }, _callee19, null, [[0, 5]]);
  }));

  return function (_x53, _x54, _x55) {
    return _ref19.apply(this, arguments);
  };
}())];

var worker = /*#__PURE__*/msw.setupWorker.apply(void 0, handlers);

var wait = function wait(ms) {
  return new Promise(function (res) {
    return setTimeout(res, ms);
  });
};

var callWithRetry = /*#__PURE__*/function () {
  var _ref = /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee(fn, depth) {
    return runtime_1.wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            if (depth === void 0) {
              depth = 0;
            }

            _context.prev = 1;
            _context.next = 4;
            return fn();

          case 4:
            return _context.abrupt("return", _context.sent);

          case 7:
            _context.prev = 7;
            _context.t0 = _context["catch"](1);

            if (!(depth > 7)) {
              _context.next = 11;
              break;
            }

            throw _context.t0;

          case 11:
            _context.next = 13;
            return wait(Math.pow(2, depth) * 10);

          case 13:
            return _context.abrupt("return", callWithRetry(fn, depth + 1));

          case 14:
          case "end":
            return _context.stop();
        }
      }
    }, _callee, null, [[1, 7]]);
  }));

  return function callWithRetry(_x, _x2) {
    return _ref.apply(this, arguments);
  };
}();

var startWorker = /*#__PURE__*/function () {
  var _ref2 = /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee3() {
    return runtime_1.wrap(function _callee3$(_context3) {
      while (1) {
        switch (_context3.prev = _context3.next) {
          case 0:
            _context3.next = 2;
            return callWithRetry( /*#__PURE__*/function () {
              var _startWorker = _asyncToGenerator( /*#__PURE__*/runtime_1.mark(function _callee2() {
                var response, data;
                return runtime_1.wrap(function _callee2$(_context2) {
                  while (1) {
                    switch (_context2.prev = _context2.next) {
                      case 0:
                        worker.start();
                        _context2.next = 3;
                        return wait(100);

                      case 3:
                        _context2.next = 5;
                        return fetch('/api/status');

                      case 5:
                        response = _context2.sent;
                        _context2.next = 8;
                        return response.json();

                      case 8:
                        data = _context2.sent;

                        if (!(data.status !== 'ok')) {
                          _context2.next = 11;
                          break;
                        }

                        throw new Error();

                      case 11:
                        return _context2.abrupt("return");

                      case 12:
                      case "end":
                        return _context2.stop();
                    }
                  }
                }, _callee2);
              }));

              function startWorker() {
                return _startWorker.apply(this, arguments);
              }

              return startWorker;
            }());

          case 2:
          case "end":
            return _context3.stop();
        }
      }
    }, _callee3);
  }));

  return function startWorker() {
    return _ref2.apply(this, arguments);
  };
}();

exports.startWorker = startWorker;
exports.worker = worker;
